getStatusFromError crashes with Vitest expect.soft

### What happened?


When `allure-vitest/setup` is enabled and a test has `expect.soft` assertions failures,
Then instead of collecting all soft failures and failing the test at the end, the test aborts with an internal `TypeError: Cannot read properties of undefined (reading 'name')` after the first soft error.

```
 FAIL  soft-assertions.test.js > multiple soft failures
TypeError: Cannot read properties of undefined (reading 'name')
 ❯ name node_modules/allure-js-commons/src/sdk/utils.ts:21:43
 ❯ runMatcherStep node_modules/allure-vitest/src/matchers.ts:474:11
 ❯ Proxy.<anonymous> node_modules/allure-vitest/src/matchers.ts:509:11-
 ❯ soft-assertions.test.js:4:22
      2|
      3| test('multiple soft failures', () => {
      4|   expect.soft(1 + 1).toBe(3); // fails softly
       |                      ^
      5|   expect.soft('a').toBe('b'); // should run; but aborts with allure-vitest/setup
      6| });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯
```

The second `expect.soft` assertion never runs.


### How can we reproduce it?

Replicate  issue with these two files:

`soft-assertions.test.js`

```js
import { expect, test } from 'vitest';

test('multiple soft failures', () => {
  expect.soft(1 + 1).toBe(3); // fails softly
  expect.soft('a').toBe('b'); // should run; but aborts with allure-vitest/setup
});
```

`vitest.config.js`

```js
import { defineConfig } from 'vitest/config';

export default defineConfig({
  test: {
    setupFiles: ['allure-vitest/setup'],
  },
});
```

## Steps
1. Install dependencies: `npm install vitest@4.1.10 allure-vitest@3.10.2`
2. Run `npx vitest run --config vitest.config.js`
3. See the first `AssertionError`, then a `TypeError: Cannot read properties of undefined (reading 'name')` from `getStatusFromError`; the second `expect.soft` does not execute

## Vitest behavior (without allure-vitest/setup) works

No config file needed:

```bash
npx vitest run soft-assertions.test.js
```

Both soft failures are collected as `[1/2]` and `[2/2]`; the test fails at the end after all assertions run.


### What did you expect?

With `allure-vitest/setup`, each soft failure should be recorded as a matcher step without crashing or aborting the test early.

Should collect and report both soft assertion failures and fail the test at the end, matching native `expect.soft` behavior:

```
FAIL  soft-assertions.test.js > multiple soft failures
AssertionError: expected 2 to be 3 // Object.is equality
[1/2]

FAIL  soft-assertions.test.js > multiple soft failures
AssertionError: expected 'a' to be 'b' // Object.is equality
[2/2]
```

### Environment

- Allure JS package and version: `allure-vitest@3.10.2`, `allure-js-commons@3.10.2`
- Test framework and version: `vitest@4.1.10`
- Node.js version: `24` 
- Package manager and version: `npm@10.8.1`
- OS: Multiple


### Additional context

# Suggested fix
Guard `getStatusFromError` against plain-object Vitest soft errors in [`packages/allure-js-commons/src/sdk/utils.ts`](https://github.com/allure-framework/allure-js/blob/main/packages/allure-js-commons/src/sdk/utils.ts#L12-L36):

```diff
 export const getStatusFromError = (error: Partial<Error>): Status => {
+  const ctorName = error?.constructor?.name ?? '';
   switch (true) {
     /**
      * Native `node:assert` and `chai` (`vitest` uses it under the hood) throw `AssertionError`
      * `jest` throws `JestAssertionError` instance
      * `jasmine` throws `ExpectationFailed` instance
      * `vitest` throws `Error` for extended assertions, so we look into stack
      * `codeceptjs-expect` errors have actual/expected properties or inspect method
      */
-    case /assert/gi.test(error.constructor.name):
-    case /expectation/gi.test(error.constructor.name):
+    case /assert/gi.test(ctorName):
+    case /expectation/gi.test(ctorName):
     case error.name && /assert/gi.test(error.name):
     case error.message && /assert/gi.test(error.message):
     case error.stack && /@vitest\/expect/gi.test(error.stack):
     case error.stack && /playwright\/lib\/matchers\/expect\.js/gi.test(error.stack):
     case error.stack && /codeceptjs-expect/gi.test(error.stack):
     case "matcherResult" in error:
     case "inspect" in error && typeof error.inspect === "function":
     case "actual" in error:
     case "expected" in error:
       return Status.FAILED;
     default:
       return Status.BROKEN;
   }
 };
```

--- a/packages/allure-js-commons/src/sdk/utils.ts
+++ b/packages/allure-js-commons/src/sdk/utils.ts
@@ -9,7 +9,17 @@ import type {
   SerializerReplacerFunc,
 } from "./types.js";
 
-export const getStatusFromError = (error: Partial<Error>): Status => {
+export const getStatusFromError = (error: unknown): Status => {
+  if (!error || (typeof error !== "object" && typeof error !== "function")) {
+    return Status.BROKEN;
+  }
+
+  const errorObject = error as Partial<Error> & Record<PropertyKey, unknown>;
+  const constructorName = String(errorObject?.constructor?.name ?? "");
+  const name = String(errorObject.name ?? "");
+  const message = String(errorObject.message ?? "");
+  const stack = String(errorObject.stack ?? "");
+
   switch (true) {
     /**
      * Native `node:assert` and `chai` (`vitest` uses it under the hood) throw `AssertionError`
@@ -18,17 +28,17 @@ export const getStatusFromError = (error: Partial<Error>): Status => {
      * `vitest` throws `Error` for extended assertions, so we look into stack
      * `codeceptjs-expect` errors have actual/expected properties or inspect method
      */
-    case /assert/gi.test(error.constructor.name):
-    case /expectation/gi.test(error.constructor.name):
-    case error.name && /assert/gi.test(error.name):
-    case error.message && /assert/gi.test(error.message):
-    case error.stack && /@vitest\/expect/gi.test(error.stack):
-    case error.stack && /playwright\/lib\/matchers\/expect\.js/gi.test(error.stack):
-    case error.stack && /codeceptjs-expect/gi.test(error.stack):
-    case "matcherResult" in error:
-    case "inspect" in error && typeof error.inspect === "function":
-    case "actual" in error:
-    case "expected" in error:
+    case /assert/gi.test(constructorName):
+    case /expectation/gi.test(constructorName):
+    case /assert/gi.test(name):
+    case /assert/gi.test(message):
+    case /@vitest\/expect/gi.test(stack):
+    case /playwright\/lib\/matchers\/expect\.js/gi.test(stack):
+    case /codeceptjs-expect/gi.test(stack):
+    case "matcherResult" in errorObject:
+    case "inspect" in errorObject && typeof errorObject.inspect === "function":
+    case "actual" in errorObject:
+    case "expected" in errorObject:
       return Status.FAILED;
     default:
       return Status.BROKEN;

--- a/packages/allure-js-commons/test/sdk/utils.spec.ts
+++ b/packages/allure-js-commons/test/sdk/utils.spec.ts
@@ -121,6 +121,12 @@ describe("allureLabelRegexp", () => {
 });
 
 describe("getStatusFromError", () => {
+  describe.each([null, undefined, "assertion error"])("with non error-like value %s", (err) => {
+    it("returns broken", () => {
+      expect(getStatusFromError(err)).toBe(Status.BROKEN);
+    });
+  });
+
   describe("with node assert error", () => {
     it("returns failed", () => {
       try {
@@ -186,6 +192,28 @@ describe("getStatusFromError", () => {
       } catch (err) {
         expect(getStatusFromError(err as Error)).toBe(Status.FAILED);
       }
+    });
+  });
+
+  describe("with a null-prototype error with matcherResult field", () => {
+    it("returns failed", () => {
+      const err = Object.assign(Object.create(null), {
+        matcherResult: {},
+        message: "expected 2 to be 3",
+      });
+
+      expect(getStatusFromError(err)).toBe(Status.FAILED);
+    });
+  });
+
+  describe("with a null-prototype error with actual and expected fields", () => {
+    it("returns failed", () => {
+      const err = Object.assign(Object.create(null), {
+        actual: 2,
+        expected: 3,
+      });
+
+      expect(getStatusFromError(err)).toBe(Status.FAILED);
     });
   });
 

--- a/packages/allure-vitest/test/spec/expect.test.ts
+++ b/packages/allure-vitest/test/spec/expect.test.ts
@@ -147,6 +147,46 @@ describe("expect", () => {
       ]);
     });
 
+    it("should report multiple soft assertion failures as matcher steps", async () => {
+      const { tests } = await runVitestInlineTest({
+        "vitest.config.ts": configFileAccessor,
+        "sample.test.ts": `
+    import { test, expect } from "vitest";
+
+    test("multiple soft failures", () => {
+      expect.soft(1 + 1).toBe(3);
+      expect.soft("a").toBe("b");
+    });
+
+  `,
+      });
+
+      expect(tests).toHaveLength(1);
+      expect(tests[0].steps).toHaveLength(2);
+      expect(tests).toMatchObject([
+        {
+          name: "multiple soft failures",
+          status: "failed",
+          steps: [
+            {
+              name: "expect(2).toBe(3)",
+              status: "failed",
+              statusDetails: {
+                message: expect.stringContaining("expected 2 to be 3"),
+              },
+            },
+            {
+              name: 'expect("a").toBe("b")',
+              status: "failed",
+              statusDetails: {
+                message: expect.stringContaining("expected 'a' to be 'b'"),
+              },
+            },
+          ],
+        },
+      ]);
+    });
+
     it("should add actual and expected values when using expect with match object", async () => {
       const { tests } = await runVitestInlineTest({
         "vitest.config.ts": configFileAccessor,

